### PR TITLE
feat(fx-2565): attribution class artworks filter

### PR DESCRIPTION
**Pull request by dzucconi**
The type of this PR is: **Feature**

This PR resolves [FX-2565]

### Description

One slightly weird thing to note here is that this counts as a single filter, which it is; but I sort of had to fight against the way actions for the filters are structured to implement it because we wanted the multi-select.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[FX-2565]: https://artsyproduct.atlassian.net/browse/FX-2565

### CODE REVIEW

**Comment by dzucconi** [comment]
As always, @MounirDhahri, let me know if I'm doing something dumb 😄

File: src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        })
+
+        return next
+      })
+    }
+  }
+
+  const filterOptions = ATTRIBUTION_CLASS_OPTIONS.map((option) => {
+    return { ...option, paramValue: nextOptions.includes(option.paramValue as string) }
+  })
```
Comment (dzucconi) [inline src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx:99]: Translates between an array of strings, which is how it's serialized for the API; to booleans for the multi-select

File: src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  const selectedAttributionClassOptions = selectedOptions.find((option) => {
+    return option.paramName === FilterParamName.attributionClass
+  })
+
+  const [nextOptions, setNextOptions] = useState<string[]>(
+    (selectedAttributionClassOptions?.paramValue as string[]) ?? []
+  )
+
+  const handleSelect = (option: FilterData, updatedValue: boolean) => {
```
Comment (dzucconi) [inline src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx:51]: Translates from the boolean to the array of strings...

File: src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx [hunk 3]
```diff
@@ -338,6 +340,7 @@ export default createPaginationContainer(
       return props.artist && props.artist.artworks
     },
     getVariables(props, { count, cursor }, fragmentVariables) {
+      console.log({ fragmentVariables })
```
Comment (mzikherman) [inline src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx:343]: 🌴

File: src/lib/utils/ArtworkFilter/FilterArtworksHelpers.ts [hunk 4]
```diff
@@ -42,6 +43,7 @@ export enum FilterDisplayName {
   // artist = "Artists",
   artistIDs = "Artists",
   artistsIFollow = "Artist",
+  attributionClass = "Rarity",
```
Comment (mzikherman) [inline src/lib/utils/ArtworkFilter/FilterArtworksHelpers.ts:46]: 👍

**Review by mzikherman — APPROVED** [review]
LGTM! (in terms of, filter parameters and such). I'll leave it to someone else to the review the 'Eigen-ness' of it 😅

**Comment by dblandin** [comment]
One Eigen-ness comment: For some of the other filters, we limit the filter options available based on the result set, as determined through the query aggregations. I don't see CLASSIFICATION listed alongside `[COLOR, DIMENSION_RANGE, MAJOR_PERIOD, MEDIUM, PRICE_RANGE]` so I'm assuming that we're statically including all 4 options here, which might lead to some *no result* states.

This speaks a bit to the difference between the filtering strategy in Eigen and what we currently have on web. I think we want to move Web closer to the Eigen strategy eventually. Curious what @MounirDhahri's take on this is.

**Comment by dzucconi** [comment]
@dblandin yeah; I find this behavior very annoying. We have the ability to do that here?

**Comment by dblandin** [comment]
> yeah; I find this behavior very annoying. We have the ability to do that here?

Generally, yes, but it would require some Gravity/Metaphysics additions. Not worth holding this PR up. I think we can follow-up with those changes. I'll create a separate ticket.

https://github.com/artsy/gravity/blob/master/app/models/search/queries/artwork_filtered_query.rb#L14-L43
https://github.com/artsy/metaphysics/blob/master/src/schema/v1/aggregations/filter_artworks_aggregation.ts

**Comment by MounirDhahri** [comment]
Yes, as Devon said, we are using aggregations to tell which filters to show and also show the available count by filter. However, in this case, I think it’s not blocking since the specs do not require us to show the count by rarity class and there are only 4 available options. Yes, it’s definitely an inferior UX but I think it’s something we can live with for now to bring value for the user, and then maybe we can make it better in a future iteration.

File: src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx [hunk 5]
```diff
@@ -353,6 +356,7 @@ export default createPaginationContainer(
         inquireableOnly: fragmentVariables.inquireableOnly,
         atAuction: fragmentVariables.atAuction,
         offerable: fragmentVariables.offerable,
+        attributionClass: fragmentVariables.attributionClass,
```
Comment (MounirDhahri) [inline src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx:359]: **nitpick:** I think this list is getting long and we should maybe change it to
```
        id: props.artist.id,
        ...fragmentVariables,
        cursor,
        count,
```

File: src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+      })
+    }
+  }
+
+  const filterOptions = ATTRIBUTION_CLASS_OPTIONS.map((option) => {
+    return { ...option, paramValue: nextOptions.includes(option.paramValue as string) }
+  })
+
+  return (
+    <>
```
Comment (MounirDhahri) [inline src/lib/Components/ArtworkFilterOptions/AttributionClassOptions.tsx:102]: **issue:** I think this is not needed

File: src/lib/Components/ArtworkFilterOptions/__tests__/AttributionClassOptions-tests.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        <AttributionClassOptionsScreen {...getEssentialProps()} />
+      </ArtworkFilterContext.Provider>
+    )
+  }
+
+  it("renders the options", () => {
+    const tree = renderWithWrappers(<MockAttributionClassOptionsScreen initialState={state} />)
+    expect(tree.root.findAllByType(OptionListItem)).toHaveLength(4)
+    const items = tree.root.findAllByType(OptionListItem)
+    expect(items.map(extractText)).toEqual(["Unique", "Limited Edition", "Open Edition", "Unknown Edition"])
```
Comment (MounirDhahri) [inline src/lib/Components/ArtworkFilterOptions/__tests__/AttributionClassOptions-tests.tsx:49]: ✨

File: src/lib/Components/ArtworkFilterOptions/__tests__/AttributionClassOptions-tests.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+      aggregations: [],
+      filterType: "artwork",
+      counts: {
+        total: null,
+        followedArtists: null,
+      },
+    }
+  })
+
+  const MockAttributionClassOptionsScreen = ({ initialState }: InitialState) => {
```
Comment (MounirDhahri) [inline src/lib/Components/ArtworkFilterOptions/__tests__/AttributionClassOptions-tests.tsx:35]: nice tests

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Looks good to me  🎉🎉

**Review by dblandin — APPROVED** [review]
LGTM. Going to go ahead and merge this in.

**Pull request merged**